### PR TITLE
Add tags to swagger specification

### PR DESCRIPTION
**Pull request by svdwoude**
Adding functionality to parse `tags` from plumber annotation and add to swagger-json.

Example:
```r
#* @apiTitle Tags Example
#* @apiTag name1 description1
#* @apiTag name2 description2
#* @apiTag name3_with_whitespaces description2


#* Example endpoint with one tag
#* @tag name1
#* @get /f1
function() {

}

#* Example endpoint with multiple tags
#* @tag name1
#* @tag name2
#* @get /f2
function() {}

#* Example endpoint with unmatched tags
#* @tag randomName
#* @get /f3
function() {}

#* Example endpoint with tag with whitespaces
#* @tag name3_with_whitespaces
#* @get /f4
function() {}
```

Which would create a swagger-ui with 4 tags:

- name1

   - /f1
   - /f2

- name2
   - /f2

- randomName
   - /f3

- name3 with whitepsace
   - /f4

Screenshot below:
plumber-tags-exmaple

### CODE REVIEW

File: R/plumber-step.R [hunk 1]
```diff
@@ -133,6 +134,9 @@ PlumberEndpoint <- R6Class(
       if (!missing(responses)){
         self$responses <- responses
       }
+      if(!missing(tags) && !is.null(tags)){
+        self$tags <- I(tags)
```
[the remark this answers is not in the document]
Comment (svdwoude) [reply R/plumber-step.R]: If we only have one tag, we want to make sure it's boxed in.
In JSON we want `tags: ["tagName"]` instead of `tags: "tagName"`

File: tests/testthat/test-globals.R [hunk 2]
```diff
@@ -28,7 +28,8 @@ test_that("parseGlobals works", {
              "#' @apiBasePath basepath",
              "#' @apiSchemes schemes",
              "#' @apiConsumes consumes",
-             "#' @apiProduces produces")
+             "#' @apiProduces produces",
+             "#' @apiTag tag description")
```
[the remark this answers is not in the document]
Comment (svdwoude) [reply tests/testthat/test-globals.R]: Multiple tags added in test.
I've also added a test to check for redundant `apiTag` definitions (= duplicates). Since swagger is pretty relaxed about unused tag definitions, I did not add a test to see wether all `apiTag` definitions are also referenced in at least one endpoint.

File: R/parse-globals.R [hunk 3]
```diff
@@ -54,6 +54,12 @@ parseOneGlobal <- function(fields, argument){
          },
          apiProduces={
            fields$produces <- strsplit(def, split="\\s+")[[1]]
+         },
+         apiTag={
+           tagMat <- stringi::stri_match(def, regex="^\\s*(\\w+)\\s+(\\S.+)\\s*$")
+           name <- gsub("_"," ",tagMat[1,2])
+           description <- tagMat[1,3]
+           fields$tags <- rbind(fields$tags,data.frame(name=name,description=description))
```
[the remark this answers is not in the document]
Comment (svdwoude) [reply R/parse-globals.R]: Good idea. Added in 1f509a0

File: R/parse-block.R [hunk 4]
```diff
@@ -179,6 +180,18 @@ parseBlock <- function(lineNum, file){
       params[[name]] <- list(desc=paramMat[1,5], type=type, required=reqd)
     }
 
+    tagMat <- stringi::stri_match(line, regex="^#['\\*]\\s*@tag\\s+(\\S.+)\\s*")
+    if (!is.na(tagMat[1,1])){
+      t <- gsub("_"," ",stri_trim_both(tagMat[1,2]))
```
[the remark this answers is not in the document]
Comment (svdwoude) [reply R/parse-block.R]: Fair enough. Updated in 8beafad

File: R/parse-globals.R [hunk 5]
```diff
@@ -54,6 +54,12 @@ parseOneGlobal <- function(fields, argument){
          },
          apiProduces={
            fields$produces <- strsplit(def, split="\\s+")[[1]]
+         },
+         apiTag={
+           tagMat <- stringi::stri_match(def, regex="^\\s*(\\w+)\\s+(\\S.+)\\s*$")
+           name <- gsub("_"," ",tagMat[1,2])
```
[the remark this answers is not in the document]
Comment (svdwoude) [reply R/parse-globals.R]: Fair enough. Updated in 8beafad

**Comment by jeffkeller87** [comment]
Really excited to see more code decoration support!

File: R/plumber-step.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (trestletech) [reply R/plumber-step.R]: Would be great to have a comment inline for that so I don't lose track of it in the future, but that makes sense.

**Comment by trestletech** [comment]
I saw the commits come in; let me know when this is ready for rereview and I'll take another pass!

**Comment by svdwoude** [comment]
Ready for review!

**Comment by trestletech** [comment]
Looks great! I pushed one more commit for some doc updates and to add you to the DESCRIPTION file. I pulled the email from your public GitHub profile, but let me know if you'd prefer to be removed or use a different address.

If that looks good, I'll go ahead and pull it in once tests pass!

**Comment by svdwoude** [comment]
Sorry about squeezing in that last typo fix.
Probably better to use the email I did my commits with: sebastiaanvandewoude@gmail.com.
Changed that meanwhile on my profile.

Thanks for checking with me :relaxed:

**Comment by trestletech** [comment]
🎉 

Thanks again for you contribution!
